cirq-ft/cirq_ft/algos/phase_estimation.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, List
+from attr import frozen
+from cirq._compat import cached_property
+import cirq
+
+from cirq_ft import infra
+
+
+@frozen
+class KitaevPhaseEstimation(infra.GateWithRegisters):
+    r"""Class representing the Kitaev Phase Estimation algorithm, originally introduced by
+    Kitaev in https://arxiv.org/abs/quant-ph/9511026."""
```

Comment:
Consider adding Neilson and Chuang as a reference, since it's a more approachable and pedagogical reference compared to the original paper.